### DISCUSSION TITLE
does custom SMTP server require SSL?

### DISCUSSION THREAD

**Discussion by vgrafe** [supabase/supabase#16500] [Questions]
Hi all!

I set up a custom SMTP server on EC2 by running [nodemailer's smtp-server](https://www.npmjs.com/package/smtp-server). For now to have a basic setup, the server just logs a message upon receiving a smtp message. The server listens to port 25, which is open in the security group.

Unfortunately, this setup won't work: looking at the logs there's an error titled "Error sending confirmation mail" with an "error" prop in the payload "Error sending confirmation email: dial tcp <edited>:25: i/o timeout".

**I did not set up any certificate on this server, nor does it expect a secure connection. Could that be the cause of my issues?**

### troubleshooting the server

Using https://www.gmass.co/smtp-test, I can send messages with success if selecting Auto, TSL or TSL when available options. Selecting SSL or None won't work.

Using a simple nodejs script, I can send messages successfully to the remote server with nodemailer, albeit only when `tls.rejectUnauthorized` is set to false.

### custom server implementation

```js
import { SMTPServer } from "smtp-server";
import { simpleParser } from "mailparser";
import { USERS } from "./env.mjs";

const server = new SMTPServer({
  host: "0.0.0.0",
  async onData(stream, session, callback) {
    let parsed;
    try {
      // Parse the raw email content
      parsed = await simpleParser(stream);

      // Log the subject and to fields
      console.log("Subject:", parsed.subject);
      console.log("To:", parsed.to.text);
    } catch (error) {
      console.error("Failed to parse message:", error);
      return callback(error);
    } finally {
      callback();
    }
  },

  onAuth(auth, session, callback) {
    if (USERS[auth.username] && USERS[auth.username] === auth.password) {
      return callback(null, { user: auth.username });
    } else {
      return callback(new Error("Invalid username or password"));
    }
  },
});

server.on("error", (err) => {
  console.log("Error %s", err.message);
});

server.listen(25);

console.log("server started!");
```

**Comment by silentworks** [comment]
You are better off asking this on Stack Overflow as its more node-mailer related than anything Supabase related.

**Reply by vgrafe** [reply]
Are you sure? The observed issue does not seem to come from nodemailer, which can be used on my EC2 instance. To me it looks like the service that supabase uses to send the SMTP message might have some undocumented requirement causing my observations.

**Comment by vgrafe** [comment]
I could not get to the bottom of this - looking closer to Supabase's codebase would probably help; I ended up adding a certificate with letsencrypt, which solved the issue. But I think there's a way to make this work with basic auth only.

**Reply by rawars** [reply]
Friend, how did you create and add the certificate?

**Reply by vgrafe** [reply]
Using https://letsencrypt.org/'s acme certbot, following the instructions did it.
